src/plugins/auto/src/cumulative_schedule.cpp

```diff
@@ -10,6 +10,21 @@
 // ------------------------------CumuSchedule----------------------------
 namespace ov {
 namespace auto_plugin {
+std::string CumuSchedule::schedule_to_next_device(const std::vector<DeviceInformation>& devices,
+                                                  std::size_t current_device_index) {
+    std::lock_guard<std::mutex> lock(m_context->m_mutex);
+    m_n_ctput_schedule_next_device =
+        m_n_ctput_schedule_next_device >= devices.size() ? 0 : m_n_ctput_schedule_next_device;
+    auto selected_device_name = devices[m_n_ctput_schedule_next_device].device_name;
+    auto schedule_policy = m_context->m_schedule_policy;
+    if (schedule_policy == ov::intel_auto::SchedulePolicy::ROUND_ROBIN) {
+        m_n_ctput_schedule_next_device++;
+    } else if (schedule_policy == ov::intel_auto::SchedulePolicy::DEVICE_PRIORITY) {
+        selected_device_name = devices[current_device_index].device_name;
```

Comment:
one hint: for device prority we don't need lock? can we optimize?

Reply:
you mean we can remove lock here. right?

Reply:
Updated.